internal/controller/dscinitialization/servicemesh.go

```diff
@@ -0,0 +1,229 @@
+	log := logf.FromContext(ctx)
+
+	if dscInit.Spec.ServiceMesh == nil {
+		log.Info("ServiceMesh not defined in DSCI, deleting ServiceMesh CR if present")
+		return r.deleteServiceMesh(ctx)
+	}
+
+	conditions := dscInit.Status.Conditions
+	switch dscInit.Spec.ServiceMesh.ManagementState {
+	case operatorv1.Managed:
+		log.Info("ServiceMesh set to Managed in DSCI, attempting to update ServiceMesh CR")
+		desiredServiceMesh := &serviceApi.ServiceMesh{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: serviceApi.ServiceMeshInstanceName,
+			},
+			Spec: serviceApi.ServiceMeshSpec{
+				ControlPlane: serviceApi.ServiceMeshControlPlaneSpec{
+					Name:              dscInit.Spec.ServiceMesh.ControlPlane.Name,
+					Namespace:         dscInit.Spec.ServiceMesh.ControlPlane.Namespace,
+					MetricsCollection: dscInit.Spec.ServiceMesh.ControlPlane.MetricsCollection,
+				},
+				Auth: serviceApi.ServiceMeshAuthSpec{
+					Namespace: dscInit.Spec.ServiceMesh.Auth.Namespace,
+					Audiences: dscInit.Spec.ServiceMesh.Auth.Audiences,
+				},
+			},
+		}
+
+		if err := controllerutil.SetControllerReference(dscInit, desiredServiceMesh, r.Client.Scheme()); err != nil {
+			return err
+		}
+
+		err := resources.Apply(
+			ctx,
+			r.Client,
+			desiredServiceMesh,
+			client.FieldOwner(fieldManager),
+			client.ForceOwnership,
+		)
+		if err != nil && !k8serr.IsAlreadyExists(err) {
+			return err
+		}
+	case operatorv1.Unmanaged:
+		log.Info("ServiceMesh set as Unmanaged in DSCI, no actions performed")
+		status.SetCondition(
+			&conditions,
+			status.CapabilityServiceMesh,
+			status.UnmanagedReason,
+			"ServiceMesh is Unmanaged in DSCI",
+			metav1.ConditionFalse,
+		)
+		status.SetCondition(
+			&conditions,
+			status.CapabilityServiceMeshAuthorization,
+			status.UnmanagedReason,
+			"ServiceMesh is Unmanaged in DSCI, ServiceMesh Authorization is therefore also Unmanaged",
+			metav1.ConditionFalse,
+		)
+		dscInit.Status.SetConditions(conditions)
+
+		if err := resources.ApplyStatus(
+			ctx,
+			r.Client,
+			dscInit,
+			client.FieldOwner(fieldManager),
+			client.ForceOwnership,
+		); err != nil {
+			log.Error(err, "failed to update DSCI status condition for ServiceMesh")
+			return err
+		}
+	case operatorv1.Removed:
+		log.Info("ServiceMesh set as Removed in DSCI, deleting ServiceMesh CR instance")
+		if err := r.deleteServiceMesh(ctx); err != nil {
+			return err
+		}
+
+		status.SetCondition(
+			&conditions,
+			status.CapabilityServiceMesh,
+			status.RemovedReason,
+			"ServiceMesh is Removed in DSCI",
+			metav1.ConditionFalse,
+		)
+		status.SetCondition(
+			&conditions,
+			status.CapabilityServiceMeshAuthorization,
+			status.RemovedReason,
+			"ServiceMesh is Removed in DSCI, ServiceMesh Authorization is therefore also Removed",
+			metav1.ConditionFalse,
+		)
+		dscInit.Status.SetConditions(conditions)
+
+		if err := resources.ApplyStatus(
```

Comment:
I won't use ApplyStatus here, since in the other part of the code the patter is to use a status update with retry, to avoid conflicts I'd use the same machinery.